crates/chain-state/src/in_memory.rs

```diff
@@ -693,6 +682,29 @@ impl BlockState {
     pub fn append_parent_chain<'a>(&'a self, chain: &mut Vec<&'a Self>) {
         chain.extend(self.parent_state_chain());
     }
+
+    /// Returns an iterator over the chain of blocks in memory.
+    pub const fn iter(self: Arc<Self>) -> BlockStateIter {
+        BlockStateIter { current: Some(self) }
+    }
+}
+
+/// An iterator over the atomically captured chain of in memory blocks.
+///
+/// This yields the blocks from newest to oldest (highest to lowest).
+#[derive(Debug, Clone)]
+pub struct BlockStateIter {
+    current: Option<Arc<BlockState>>,
+}
+
+impl Iterator for BlockStateIter {
+    type Item = Arc<BlockState>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let current = self.current.take()?;
+        self.current = current.parent.clone();
+        Some(current)
+    }
 }
```

Comment:
This is exactly the impl of std::iter::successors
```rust
std::iter::successors(Some(self), |state| state.parent.clone())
```

Reply:
yes, but I prefer having this

Reply:
actually I changed my mind, the docs in iter should be sufficient